PracticeGit/Test.c

```diff
@@ -1,13 +1,151 @@
+void GetUserInput ();
+
+/// <summary>Compare Arrays returns true or false. </summary>
+bool ArraysEqual (int arr1[], int arr2[], int size);
+
+typedef struct {
+   int array[MAXSIZE], size;
+} TestCase;
+
+void PrintArray (int array[], int size) {
+   for (int i = 0; i < size; i++) printf ("%3d", array[i]);
+}
+
+bool ArraysEqual (int arr1[], int arr2[], int size) {
+   for (int i = 0; i < size; i++) {
+      if (arr1[i] != arr2[i]) return false;
+   }
+   return true;    // Arrays are equal if all elements match
+}
+
+void RunQuickSortTest () {
+   TestCase testCases[] = {
+       {{20, 49, 11, 3, 89, 75, 11, 12, 1}, 9},
+       {{2, 2, 2, 2, 2, 2, 2}, 7},
+       {{9}, 1},
+       {{3, -1, -7, 5, 2, 0, -7}, 7},
+       {{1, 2, 3, 4, 5, 6}, 6},
+       {{0, -1, -2, -3, -4, -5}, 6},
+       {{36, 69}, 2},
+       {{22, 35, 32, 39, 46, 54, 55}, 7},
+       {{12, 13, 14, 15, 16, 11}, 6}
+   };
+   int expectedResults[][MAXSIZE] = {
+       {1, 3, 11, 11, 12, 20, 49, 75, 89},
+       {2, 2, 2, 2, 2, 2, 2},
+       {9},
+       {-7, -7, -1, 0, 2, 3, 5},
+       {1, 2, 3, 4, 5, 6},
+       {-5, -4, -3, -2, -1, 0},
+       {36, 69},
+       {22, 32, 35, 39, 46, 54, 55},
+       {11, 12, 13, 14, 15, 16}
+   };
+   int elementSearch[] = { 11, 1, 0, -1, 2, 6, 36, 53, 11 };
+   int testCount = sizeof (testCases) / sizeof (testCases[0]);
+   printf ("+--------------------------------------------------+---------------+------------------+--------+\n"
+      "|                Arrays                            | Key to Search | Index of element | Result |\n"
+      "+--------------------------------------------------+---------------+------------------+--------+\n");
+   for (int i = 0; i < testCount; i++) {
+      int size = testCases[i].size, * arrayToSort = testCases[i].array;
+      printf ("| Unsorted Array   | ");
+      PrintArray (arrayToSort, size);
+      printf ("%*s|", (MAXSIZE - size) * 3, " ");
+      int searchValue = elementSearch[i];   // Element to search for in the current test case
+      QuickSort (arrayToSort, 0, size - 1);
+      int foundIndex = BinarySearch (arrayToSort, size, searchValue);
+      if (foundIndex != ELEMENT_NOT_FOUND) printf ("    %-11d| Found at index %-d |", searchValue, foundIndex);
+      else printf ("    %-11d|    Not Found     |", searchValue);
+      printf ("  %-8s  |\n", ArraysEqual (arrayToSort, expectedResults[i], size) ? GREEN"Pass"RESET : RED"Fail"RESET);
+      printf ("| Sorted Array     | ");
+      PrintArray (arrayToSort, size);
+      printf ("%*s|", (MAXSIZE - size) * 3, " ");
+      printf ("\n+--------------------------------------------------+---------------+------------------+--------+\n");
+   }
+}
 
-#include "Program.h"
+void GetUserInput () {
+   int array[MAXSIZE], size;
+   char input[256];
+   printf ("Enter the number of elements (1 to %d): ", MAXSIZE);
+   fgets (input, sizeof (input), stdin);
+   size = atoi (input);     // Convert input to an integer
```

Comment:
What if the input is an alphabet? The value of size becomes 0 and the code works fine since you expect for an input between 1 and 10. But the invalidity for input, say 'a', is not because its integer equivalent is zero, but for the fact that it is not an integer in itself. So, try to have a check to verify whether the input is a valid number first before converting into its integer equivalent.